msctl

```diff
@@ -1295,14 +1295,14 @@ syncMirrorImage() {
 # @param 1 The world server to connect to.
 # ---------------------------------------------------------------------------
 serverConsole() {
-  local LINE WORLD_DIR
+  local CHAR WORLD_DIR
   # Make sure that the world's directory exists.
   WORLD_DIR="$WORLDS_LOCATION/$1"
   # Follow the console output buffer file.
   tail -f --pid=$$ $WORLD_DIR/console.out &
   # Copy user input to the console input buffer file.
-  while read LINE; do
-    echo "$LINE" >>$WORLD_DIR/console.in
+  while read -r -n1 CHAR; do
```

Comment:
Well, it turns out that `-n` is not a valid option in the dash version of read.  It works great in bash though!  Back to the drawing board on this patch..

Reply:
I found this bit on [stackoverflow](https://stackoverflow.com/questions/8725925/how-to-read-just-a-single-character-in-shell-script):
```
read_char() {
  stty -icanon -echo
  eval "$1=\$(dd bs=1 count=1 2>/dev/null)"
  stty icanon echo
}
```

I'll try that out later.

Reply:
This is a tough problem.

This is where I am at:
```patch
diff --git a/msctl b/msctl
index 2289cfb..6dd5b6a 100755
--- a/msctl
+++ b/msctl
@@ -1289,6 +1289,12 @@ syncMirrorImage() {
   fi
 }

+read_char() {
+  stty raw -echo
+  eval "$1=\$(dd bs=1 count=1 2>/dev/null)"
+  stty -raw echo
+}
+
 # ---------------------------------------------------------------------------
 # Display the server console.
 #
@@ -1301,7 +1307,18 @@ serverConsole() {
   # Follow the console output buffer file.
   tail -f --pid=$$ $WORLD_DIR/console.out &
   # Copy user input to the console input buffer file.
-  while read -r -n1 CHAR; do
+  while read_char CHAR; do
+    # Capture Ctrl+D and exit if found.
+    if [ "$CHAR" = "$(printf "\004")" ]; then
+      exit
+    fi
+    # Fix backspace character.
+    if [ "$CHAR" = "$(printf "\177")" ]; then
+      CHAR="\010\040\010"
+    fi
+    # Echo user input.
+    printf "$CHAR"
+    # Print input to console.in.
     printf "$CHAR" >> $WORLD_DIR/console.in
   done
 }
````

I've worked around an issue with the backspace character, and I now capture Ctrl+D and exit correctly.  There may be other issues though!

More work to come.